Objects/descrobject.c

```diff
@@ -221,7 +223,7 @@ descr_setcheck(PyDescrObject *descr, PyObject *obj, PyObject *value,
 static int
 member_set(PyMemberDescrObject *descr, PyObject *obj, PyObject *value)
 {
-    int res;
+    int res = -1;
 
     if (descr_setcheck((PyDescrObject *)descr, obj, value, &res))
```

Comment:
I don't understand the purpose of the res variable in descr_setcheck(), it is always set to -1 when descr_setcheck() result is non-zero. Maybe remove res and return -1 directly in the caller (here)?